src/ocaml_flags.ml

```diff
@@ -5,47 +5,48 @@ open Build.O
 let default_ocamlc_flags   = ["-g"]
 let default_ocamlopt_flags = ["-g"]
 
-let dev_mode_warnings ~dune_version =
-  let codes =
-    [ 4
-    ; 29
-    ; 40
-    ; 41
-    ; 42
-    ; 44
-    ; 45
-    ; 48
-    ; 58
-    ; 59
-    ; 60
-    ]
-  in
-  let codes =
-    if dune_version >= (1, 11) then
-      codes @ [66]
-    else
-      codes
+let dev_mode_warnings =
+  (* New warnings should be introduced here *)
+  let all =
+    Int.Set.diff
+      (Int.Set.of_list (List.init 62 ~f:succ))
+      (Int.Set.of_list
+         [ 4
+         ; 29
+         ; 40
+         ; 41
+         ; 42
+         ; 44
+         ; 45
+         ; 48
+         ; 58
+         ; 59
+         ; 60
+         ]) in
+  let by_version =
+    match Syntax.Version.Per.of_mapping [] ~default:all with
+    | Ok s -> s
+    | Error (_, _, _) -> assert false
   in
-  "@a" ^
-  String.concat ~sep:""
-    (List.map ~f:(sprintf "-%d") codes)
-
-let default_warnings =
-  "-40"
+  fun ~dune_version ->
+    Syntax.Version.Per.get by_version dune_version
+    |> Int.Set.to_list
+    |> List.map ~f:(sprintf "@%d")
+    |> String.concat ~sep:""
```

Comment:
I just noticed that we can pass intervals to the compiler, for instance: `@1..3@4..28`. We should do that here to reduce the length of the command line.

Reply:
Good call. I did that.